package/MDAnalysis/core/groups.py

```diff
@@ -201,10 +197,19 @@ def _subclass(cls, singular=None):
         type
             A subclass of :class:`_TopologyAttrContainer`, with the same name.
         """
-        if singular is not None:
-            return type(cls.__name__, (cls,), {'_singular': bool(singular)})
+        newcls = type(cls.__name__, (cls,), {'_singular': bool(singular)})
+        if singular:
+            newcls._SETATTR_WHITELIST = {
+                'position', 'velocity', 'force', 'dimensions',
+                'atoms', 'residue', 'residues', 'segment', 'segments',
+            }
         else:
-            return type(cls.__name__, (cls,), {})
+            newcls._SETATTR_WHITELIST = {
+                'positions', 'velocities', 'forces', 'dimensions',
+                'atoms', 'residue', 'residues', 'segment', 'segments',
+            }
```

Comment:
Whitelist is the same in both cases. What is the purpose of singular in this particular case?

Reply:
one version is singular (position, velocity, etc) one is plural (position**s**, veloctiti**es**)

Reply:
why are the segments then plural? How can something have one position and multiple segments or atoms?

Reply:
yeah you're right - fixed